### PR TITLE
Fix GitLab CI

### PR DESCRIPTION
**Pull request by SamuelRiedel**
Specify Halide's library path while installing since the default differs between machines.

### Fixed
- Specify Halide's library path while installing

## Checklist

- [x] Automated tests pass
- [x] Changelog updated
- [x] Code style guideline is observed

### CODE REVIEW

**Comment by SamuelRiedel** [comment]
Halide now also works internally, but Verilator fails as expected. @suehtamacv feel free to add potential fixes we need for Verilator to this PR as well.

**Comment by suehtamacv** [comment]
> Halide now also works internally, but Verilator fails as expected. @suehtamacv feel free to add potential fixes we need for Verilator to this PR as well.

Done! :tada:

File: .gitlab/.gitlab-ci.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ variables:
   APPS: "tests"
   RUST_LOG: 'memora=debug'
   VERILATOR_ROOT: '$CI_PROJECT_DIR/install/verilator'
+  CLANG_PATH: '/usr/scratch/riseten/matheusd/llvm/'
```
Comment (suehtamacv) [inline .gitlab/.gitlab-ci.yml:13]: Is this fine for the time being?

File: Makefile [hunk 2]
```diff
@@ -35,9 +35,22 @@ CXX = g++
...
-# We need a recent LLVM installation (>11) to compile Verilator
-CLANG_CC  ?= clang
-CLANG_CXX ?= clang++
+# We need a recent LLVM installation (>11) to compile Verilator.
+# We also need to link the binaries with LLVM's libc++.
+# Define CLANG_PATH to be the path of your Clang installation.
+# At IIS, check .gitlab/.gitlab-ci.yml for an example CLANG_PATH.
+
+ifneq (${CLANG_PATH},)
+        CLANG_CC       := $(CLANG_PATH)/bin/clang
```
Comment (suehtamacv) [inline Makefile:44]: If you define `CLANG_PATH`, then you do not need to specify `CLANC_CC`. What do you think about this solution?

File: .gitlab/.gitlab-ci.yml [hunk 3]
```diff
@@ -189,10 +190,7 @@ apps-halide:
     - git submodule update --init --recursive -- hardware/deps/*
     - make patch-hw
     - cd hardware
-    - |
```
Comment (suehtamacv) [inline .gitlab/.gitlab-ci.yml:192]: Should we add a matrix as well, as we do on GitHub, or is this redundant?

**Review by suehtamacv — COMMENTED** [review]
Thanks, Samuel! The CI now works on GitLab as well. What do you think of my changes?

File: .gitlab/.gitlab-ci.yml (hunk 1 above)
Comment (SamuelRiedel) [reply .gitlab/.gitlab-ci.yml:13]: Yes

File: .gitlab/.gitlab-ci.yml (hunk 3 above)
Comment (SamuelRiedel) [reply .gitlab/.gitlab-ci.yml:192]: I don't think it's necessary.

File: Makefile (hunk 2 above)
Comment (SamuelRiedel) [reply Makefile:44]: Perfect.

Just a small consistency question. How many spaces are we using for this kind of indentation? Here you use eight, above we have zero. Personally, I think if we want to indent the blocks to make them more readable, we should use two spaces as we do in all the other source files.

File: hardware/Makefile [hunk 4]
```diff
@@ -142,6 +142,15 @@ VERILATOR_FLAGS += $(VERILATOR_WAIVE)
 # VERILATOR_FLAGS += --trace --trace-fst --trace-structs --trace-params --trace-max-array 1024
 # VERILATOR_FLAGS += --debug
 
+# We need to link the verilated model against LLVM's libc++.
+# Define CLANG_PATH to be the path of your Clang installation.
+# At IIS, check .gitlab/.gitlab-ci.yml for an example CLANG_PATH.
+
+ifneq (${CLANG_PATH},)
+        VERILATOR_FLAGS += -CFLAGS "-nostdinc++ -isystem $(CLANG_PATH)/include/c++/v1"
```
Comment (SamuelRiedel) [inline hardware/Makefile:150]: The same here.

**Review by SamuelRiedel — COMMENTED** [review]
Thanks, LGTM for now. Just the question about the indentation.

File: Makefile (hunk 2 above)
Comment (suehtamacv) [reply Makefile:44]: It is not consistent at all. In the `hardware`'s Makefile, we indent with two spaces.

**Review by suehtamacv — APPROVED** [review]

**Pull request merged**